reporting/bufferSize.html

```diff
@@ -8,22 +8,20 @@
 <script src="/resources/testdriver-vendor.js"></script>
 <script>
   // Test the buffer size (100) of ReportingObserver.
-  async_test(async function(test) {
+  promise_test(async function(test) {
     for (i = 0; i != 110; ++i)
       await test_driver.generate_test_report("" + i);
 
-    var observer = new ReportingObserver(function(reports) {
-      test.step(function() {
-        // Only (the most recent) 100 reports should be observed, even though
-        // 110 were buffered.
-        assert_equals(reports.length, 100);
-        for(i = 0; i != 100; ++i) {
-          assert_equals(reports[i].body.message, "" + (i + 10));
-        }
-      });
-
-      test.done();
-    }, {buffered: true});
-    observer.observe();
+    return new Promise(resolve => {
```

Comment:
Wait, you can grab the results of the `resolve` too? :O

That's super cool, I wonder if we should update https://web-platform-tests.org/writing-tests/testharness-api.html?highlight=promise_test#promise-tests (yet again) to suggest `await`-ing rather than `then`-ing for this trick. (see the 'Note that in the promise chain constructed ...' section)

Reply:
I would definitely suggest updating those docs. I get the impression they were written before async/await were widespread enough to be depended upon.